php/php-8.2.0/patches/0020-Avoid-syslog-setjmp-longjmp-in-libgd.patch

```diff
@@ -0,0 +1,160 @@
+From 76d860faf9a37759c7f751ec89f909c4d06ee77b Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Jesu=CC=81s=20Gonza=CC=81lez?= <jesusgm@vmware.com>
+Date: Tue, 14 Feb 2023 19:09:34 +0100
+Subject: [PATCH] fix: Avoid `#include <syslog.h>` since it is not part of
+ wasi-libc
+
+---
+ ext/gd/libgd/gd_errors.h | 12 +++++++-----
+ 1 file changed, 7 insertions(+), 5 deletions(-)
+
+diff --git a/ext/gd/libgd/gd_errors.h b/ext/gd/libgd/gd_errors.h
+index 8a28bf6051..e84cef4578 100644
+--- a/ext/gd/libgd/gd_errors.h
++++ b/ext/gd/libgd/gd_errors.h
+@@ -1,11 +1,13 @@
+ #ifndef GD_ERRORS_H
+ #define GD_ERRORS_H
+ 
+-#ifndef _WIN32
+-# include <syslog.h>
+-#else
+-# include "win32/syslog.h"
+-#endif
++#ifndef __wasi__
++    #ifndef _WIN32
++    # include <syslog.h>
++    #else
++    # include "win32/syslog.h"
++    #endif
++#endif /* __wasi__ */
```

Comment:
@ereslibre I like your suggestion.
This is my reasoning behind this: if we were submitting this patch to PHP, I'll do it as you suggest. But since this is a patch over the original source code only for WLR, wouldn't you leave it this way for clarity on what are we changing only for this use case?

Reply:
My take is that it's all the same in this context. Because this is internal, as @gzurl said.
Also, the effort to do it properly would be to checkout and regenerate the whole patch.

Reply:
+1, let's keep it as it is